compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -48,13 +48,16 @@ object Inliner {
       def accessorNameKind = InlineAccessorName
 
       /** A definition needs an accessor if it is private, protected, or qualified private
-        *  and it is not part of the tree that gets inlined. The latter test is implemented
-        *  by excluding all symbols properly contained in the inlined method.
-        */
+       *  and it is not part of the tree that gets inlined. The latter test is implemented
+       *  by excluding all symbols properly contained in the inlined method.
+       *
+       *  Constants don't need accessors since they are inlined in FirstTransform.
+       */
       def needsAccessor(sym: Symbol)(implicit ctx: Context) =
         sym.isTerm &&
         (sym.is(AccessFlags) || sym.privateWithin.exists) &&
-        !sym.isContainedIn(inlineSym)
+        !sym.isContainedIn(inlineSym) &&
+        !sym.info.widenTermRefExpr.isInstanceOf[ConstantType]
```

Comment:
Tou also need to check `isIdempotentExpr` like we do in `TreeInfo#constToLiteral` which is called from `FirstTransform`. Alternatively you could just check `is(Final)` I think.

Reply:
Right so I'd check is(Final)

Reply:
Can you elaborate why `is(Final)`?
```scala
private final val x: Int = 1 // won't be inlined by FirstTransform, so need accessor
private final def y: Int = 1 // won't be inlined either, need accesor
```

Reply:
I think the right test is `
```scala
!(sym.isStable && sym.info.widenTermRefExpr.isInstanceOf[ConstantType])
```
That excludes `defs` with constant result. These have to be called anyway for their side effects.